### PR TITLE
Fix autowiring of API action classes

### PR DESCRIPTION
**Pull request by dontub**
See comment of class `ActionPropertyAutowireFixPass` for details.

### CODE REVIEW

File: Civi/Civioffice/DependencyInjection/Compiler/ActionPropertyAutowireFixPass.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+namespace Civi\Civioffice\DependencyInjection\Compiler;
+
+use Civi\Api4\Generic\AbstractAction;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Symfony DI (v5 and v6) tries to autowire properties annotated with @required.
+ * Though CiviCRM action parameters can be annotated in this way to make them
```
Comment (jensschuppe) [inline Civi/Civioffice/DependencyInjection/Compiler/ActionPropertyAutowireFixPass.php:12]: `s/CiviCRM action/CiviCRM API action`

File: Civi/Civioffice/DependencyInjection/Compiler/ActionPropertyAutowireFixPass.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * mandatory. This pass clears the properties that are going to be injected for
+ * action classes registered as services in the Civi\Civioffice\Api4\Action
+ * namespace. (In Symfony DI v7 support of annotations is dropped in favor of
+ * PHP attributes.)
+ */
+final class ActionPropertyAutowireFixPass implements CompilerPassInterface {
```
Comment (jensschuppe) [inline Civi/Civioffice/DependencyInjection/Compiler/ActionPropertyAutowireFixPass.php:18]: `s/ActionPropertyAutowireFixPass/ApiActionPropertyAutowireFixPass`

File: Civi/Civioffice/DependencyInjection/Compiler/ActionPropertyAutowireFixPass.php
Comment (jensschuppe) [inline Civi/Civioffice/DependencyInjection/Compiler/ActionPropertyAutowireFixPass.php:1]: Let's just make it clearer that this is about CiviCRM *API* actions.

File: Civi/Civioffice/DependencyInjection/Compiler/ActionPropertyAutowireFixPass.php (hunk 2 above)
Comment (dontub) [reply Civi/Civioffice/DependencyInjection/Compiler/ActionPropertyAutowireFixPass.php]: Changed.

**Comment by dontub** [comment]
The compiler pass is now registered with the correct type.

**Pull request merged**